### ISSUE TITLE
Ndarray properties can't be used for an optional array

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#1136]
**Describe the bug**
Ndarray properties can't be used for an optional array.

**To Reproduce**
Provide code to reproduce the behaviour:
```python
def f(x : 'int[:]' = None):
    if x is not None:
        print(x.shape[0])
        x += 3
```

Provide the generated code, or the error message:
```bash/fortran/c
/home/emily/Code/test/tmp/__pyccel__/opt_arrays.c: In function ‘f’:
/home/emily/Code/test/tmp/__pyccel__/opt_arrays.c:14:26: error: ‘x’ is a pointer; did you mean to use ‘->’?
   14 |         printf("%ld\n", x.shape[0]);
      |                          ^
      |                          ->
/home/emily/Code/test/tmp/__pyccel__/opt_arrays.c:15:26: error: ‘x’ is a pointer; did you mean to use ‘->’?
   15 |         for (i = 0; i < x.shape[0]; i += 1)
      |                          ^
      |                          ->
```

**Expected behaviour**
The problem arises because the `_print` function isn't used to print the variable. This will also cause problems when #537 is fixed as we won't be able to do:
```python
a = my_func_returns_array().shape[0]
```
With the `_print` function call pyccel should generate `(*x).shape[0]` instead of `x.shape[0]`, however a solution which allows us to print `x->shape[0]` would be preferred. This may be possible with a class-wide variable but care should be taken not to break cases where multiple property accessors are handled in the same line. E.g:
```python
def f(b : 'int[:,:]' = None):
    a = b[my_func_returns_array().shape[0]].shape[0]
```

**Language**
C

**Comment by aihya** [comment]
I wonder if there exist some tests for this bug already.

**Comment by EmilyBourne** [comment]
I couldn't find one. There are lots of tests for optional arrays but they don't seem to use the ndarray properties (other than `GET_ELEMENT` which looks like a function so the `*` is correctly printed)

**Comment by aihya** [comment]
Could `x` be a stack array in the first example?

**Comment by EmilyBourne** [comment]
> Could `x` be a stack array in the first example?

No. For a variable to be allocated on the stack its size needs to be known and the function needs to be responsible for allocating it. Therefore it doesn't make much sense to apply this concept to arguments.

Let's take the example of integers in C++:
```
void f (int& my_var);

void main() {
    int stack_variable(0);
    int* heap_variable = new int(0);
    f(stack_variable);
    f(*heap_variable);
}
```

You can see that `f` doesn't care whether `my_var` is on the stack or the heap as it does not change how it is handled with the body of `f`

**Comment by aihya** [comment]
This doesn't seem to be a problem anymore, the test cases above works fine.

**Comment by EmilyBourne** [comment]
I can confirm. This was fixed by #858
